test/services/updatePsyFields.spec.js

```diff
@@ -1,27 +1,115 @@
+      expect(result).to.eql('Français');
+    });
+
+    it('should add Français if input does not contains it', async () => {
+      const result = updatePsyFields.addFrenchLanguageIfMissing('Anglais');
+      expect(result).to.eql('Français, Anglais');
+    });
+
+    it('should not add Français if input contains it', async () => {
+      const result = updatePsyFields.addFrenchLanguageIfMissing('Français, Anglais');
+      expect(result).to.eql('Français, Anglais');
+    });
+
+    it('should not add Français if input contains it (lower case)', async () => {
+      const result = updatePsyFields.addFrenchLanguageIfMissing('français, anglais');
+      expect(result).to.eql('français, anglais');
+    });
+
+    it('should not add Français if input contains it (no special char)', async () => {
+      const result = updatePsyFields.addFrenchLanguageIfMissing('francais, anglais');
+      expect(result).to.eql('francais, anglais');
+    });
 
-  it("should not add french if 'francais' is there", async () => {
-    updatePsyFields.addFrenchLanguageIfMissing('francais').should.equal('francais');
+    it('should not add Français if input contains it (no specific order)', async () => {
+      const result = updatePsyFields.addFrenchLanguageIfMissing('Espagnol Francais Anglais');
+      expect(result).to.eql('Espagnol Francais Anglais');
+    });
   });
 
-  it('should not add french (capitalized) if already there', async () => {
-    updatePsyFields.addFrenchLanguageIfMissing('Français et espagnol').should.equal('Français et espagnol');
+  describe('fields', () => {
+    let addFrenchLanguageIfMissingStub;
+    beforeEach(() => {
+      addFrenchLanguageIfMissingStub = sinon.stub(updatePsyFields, 'addFrenchLanguageIfMissing')
+        .returnsArg(0);
+    });
+
+    afterEach(() => {
+      addFrenchLanguageIfMissingStub.restore();
+    });
+
+    it('Editable psy fields should add french language', () => {
+      const psy = {
+        languages: 'random stuff',
+      };
+
+      updatePsyFields.editablePsyFields(psy);
+      sinon.assert.calledWith(addFrenchLanguageIfMissingStub, psy.languages);
+    });
+
+    it('Editable and non editable fields should not overlap', () => {
+      const psy = {
+        email: 'email',
+        address: 'adress',
+        departement: 'departement',
+        region: 'region',
+        phone: 'phone',
+        website: 'website',
+        description: 'description',
+        teleconsultation: true,
+        languages: 'francais',
+        personalEmail: 'personalEmail',
+        firstNames: 'firstNames',
+        lastName: 'lastName',
+        archived: true,
+        state: DossierState.accepte,
+        training: 'training',
+        adeli: 'adeli',
+        diploma: 'diploma',
+        dossierNumber: 'dossierNumber',
+        assignedUniversityId: 'assignedUniversityId',
+        isConventionSigned: true,
+        selfModified: true,
+        active: true,
+        inactiveUntil: 'inactiveUntil',
```

Comment:
We could add a test to ensure that these fields are not in `editablePsy` and `nonEditablePsy`.